### ISSUE TITLE
Bad mocks when using arrays of enums

### ISSUE THREAD

**Issue by arlyon** [anymaniax/orval#342]
Hi!

I am about to introduce mocking into my project, to build up some ui tests. I have noticed in a particular case orval doesn't output valid code, for which I have provided a repro using the latest master branch of orval:

## What are the steps to reproduce this issue?

`schema.json`
```json
{
  "swagger": "2.0",
  "info": {
    "version": "1.0.0",
    "title": "Swagger Petstore",
    "description": "A sample API that uses a petstore as an example to demonstrate features in the swagger-2.0 specification",
    "termsOfService": "http://swagger.io/terms/",
    "contact": {
      "name": "Swagger API Team"
    },
    "license": {
      "name": "MIT"
    }
  },
  "host": "petstore.swagger.io",
  "basePath": "/api",
  "schemes": ["http"],
  "consumes": ["application/json"],
  "produces": ["application/json"],
  "paths": {
    "/pets": {
      "get": {
        "description": "Returns the subset of pets from the system that the user has access to",
        "produces": ["application/json"],
        "responses": {
          "200": {
            "description": "A list of pets.",
            "schema": {
              "uniqueItems": true,
              "type": "array",
              "items": {
                "type": "string",
                "enum": ["Spot", "Rover", "Toby", "Archie", "Barcus Aurelius"]
              }
            }
          }
        }
      }
    }
  }
}
```

`orval.config.js`
```js
module.exports = {
  pets: {
    input: './schema.json',
    output: {
      mode: 'tags',
      client: 'react-query',
      target: '.',
      schemas: 'models',
      prettier: true,
      mock: true,
    },
  },
};
```

The specific issue crops up in handling arrays that also contain enums, which can be attributed to this chunk of code: 

https://github.com/anymaniax/orval/blob/8e99ff9c4b059bc975b437b684183e6e1ff5d0ff/src/core/getters/scalar.mock.ts#L114-L126

The relevant part of the output is as follows:

`default.ts`
```ts
export const getGetPetsMock = () =>
  [...Array(faker.datatype.number({ min: 1, max: 5 }))].reduce(
    ({ values, enums }) => {
      const newValue =
        enums[faker.datatype.number({ min: 1, max: enums.length })];
      return {
        values: [...values, newValue],
        enums: enums.filter((v: GetPets200Item[]) => newValue !== v), // probably incorrect types
      };
    },
    { values: [], enums: Object.values(GetPets200Item[]) }, // javascript error
  );
```

I don't believe the code generated on lines 120 nor on 122 are correct. Happy to write a fix with some guidance as to what could be happening here.

**Comment by anymaniax** [comment]
Will be fix in the next version

**Comment by anymaniax** [comment]
Fixed with `6.7.0`

**Comment by arlyon** [comment]
Hi!

Had a chance to test this a little deeper, and managed to find another broken case:

```json
{
  "swagger": "2.0",
  "info": {
    "version": "1.0.0",
    "title": "Swagger Petstore",
    "description": "A sample API that uses a petstore as an example to demonstrate features in the swagger-2.0 specification",
    "termsOfService": "http://swagger.io/terms/",
    "contact": {
      "name": "Swagger API Team"
    },
    "license": {
      "name": "MIT"
    }
  },
  "host": "petstore.swagger.io",
  "basePath": "/api",
  "schemes": ["http"],
  "consumes": ["application/json"],
  "produces": ["application/json"],
  "paths": {
    "/pets": {
      "get": {
        "description": "Returns the subset of pets from the system that the user has access to",
        "produces": ["application/json"],
        "responses": {
          "200": {
            "description": "A list of pets.",
            "schema": {
              "uniqueItems": true,
              "type": "object",
              "properties": {
                "name": {
                  "uniqueItems": true,
                  "type": "array",
                  "items": {
                    "type": "string",
                    "enum": ["Spot", "Rover", "Toby", "Archie", "Barcus Aurelius"]
                  }
                }
              }
            }
          }
        }
      }
    }
  }
}
```
